libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -296,6 +312,18 @@ func (p *pod) emitEvents(pod *kubernetes.Pod, flag string, containers []kubernet
 			safemapstr.Put(annotations, k, v)
 		}
 		kubemeta["annotations"] = annotations
+		if p.namespaceWatcher != nil {
+			if rawNs, ok, err := p.namespaceWatcher.Store().GetByKey(pod.Namespace); ok && err == nil {
+				if namespace, ok := rawNs.(*kubernetes.Namespace); ok {
+					defaults := common.MapStr{}
+
+					for k, v := range namespace.GetAnnotations() {
+						safemapstr.Put(defaults, k, v)
+					}
+					kubemeta["defaults"] = defaults
```

Comment:
I don't think we need to create a new key for this. metadata package is already able to put namespace annotations under `kubernetes.namespace.annotations` right? We can add that to the `kubemeta` object, in the same way we always add pod `annotations`.

Reply:
onl in 8.0 we will have `namespace` as a `MapStr`. until then it would be `namespace_annotations`

Reply:
thoughts?

Reply:
good catch, that's true! I think we should do the same here